x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTruncProcessor.java

```diff
@@ -48,9 +50,14 @@ public static Object process(Object truncateTo, Object timestamp, ZoneId zoneId)
         if (truncateTo instanceof String == false) {
             throw new SqlIllegalArgumentException("A string is required; received [{}]", truncateTo);
         }
+
+        List<String> similar = Part.findSimilar((String) truncateTo);
```

Comment:
I don't get why you moved this here. In my opinion the typos that the findSimilar is addressing are irrelevant of the edge case with `week` and interval. As a user, first you get an error message with proposed units to fix the typo, and then a new error message about the `week`.

So I would move the findSimilar where it was and instead move the week related if after line 72 where the check of the data type of the 2nd arg which seems a higher prio error than the edge case.